data_steward/tools/create_combined_backup_dataset.py

```diff
@@ -0,0 +1,535 @@
+
+WRITE_APPEND = 'WRITE_APPEND'
+WRITE_TRUNCATE = 'WRITE_TRUNCATE'
+
+
+def assert_tables_in(client: BigQueryClient, dataset_id: str):
+    """
+    Raise assertion error if any CDM tables missing from a dataset
+
+    :param client: a BigQueryClient
+    :param dataset_id: dataset to check for tables in
+    """
+    tables = client.list_tables(dataset_id)
+    table_ids = set([table.table_id for table in tables])
+    LOGGER.info(f'Confirming dataset, {dataset_id}, has tables: {tables}')
+    if table_ids != set(combine_consts.TABLES_TO_PROCESS):
+        raise RuntimeError(
+            f'Dataset, {dataset_id}, is missing tables {set(combine_consts.TABLES_TO_PROCESS) - table_ids}.  '
+            f'Aborting.')
+
+
+def assert_ehr_and_rdr_tables(client: BigQueryClient,
+                              unioned_ehr_dataset_id: str, rdr_dataset_id: str):
+    """
+    Raise assertion error if any CDM tables missing from EHR or RDR dataset
+
+    :param client: a BigQueryClient
+    :param unioned_ehr_dataset_id: Identifies unioned_ehr dataset name
+    :param rdr_dataset_id: Identifies the rdr dataset name
+
+    :return: None
+    """
+    assert_tables_in(client, unioned_ehr_dataset_id)
+    assert_tables_in(client, rdr_dataset_id)
+
+
+def create_cdm_tables(client: BigQueryClient, combined_dataset_id: str):
+    """
+    Create all CDM tables
+
+    :param client: BigQueryClient
+    :param combined_dataset_id: Identifies combined dataaset name
+
+    Note: Recreates any existing tables
+    """
+    for table in resources.CDM_TABLES:
+        LOGGER.info(f'Creating table {combined_dataset_id}.{table}...')
+        schema_list = client.get_table_schema(table_name=table)
+        dest_table = f'{client.project}.{combined_dataset_id}.{table}'
+        dest_table = bigquery.Table(dest_table, schema=schema_list)
+        table = client.create_table(dest_table)  # Make an API request.
+        LOGGER.info(f"Created table: `{table.table_id}`")
+
+
+def query(client: BigQueryClient,
+          q: str,
+          dst_dataset_id: str = None,
+          dst_table_id: str = None,
+          write_disposition: str = None):
+    """
+    Run query and block until job is done
+
+    :param client: a BigQueryClient
+    :param q: SQL statement
+    :param dst_dataset_id: if set, output is saved in a dataset with the specified id
+    :param dst_table_id: if set, output is saved in a table with the specified id
+    :param write_disposition: WRITE_TRUNCATE, WRITE_EMPTY, or WRITE_APPEND (default, to preserve schema)
+    """
+    job_config = bigquery.job.QueryJobConfig()
+    if dst_table_id and dst_dataset_id:
+        job_config.destination = f'{client.project}.{dst_dataset_id}.{dst_table_id}'
+        job_config.write_disposition = write_disposition
+
+    prefix = f'cb_{dst_table_id if dst_table_id else ""}_{datetime.now().strftime("%m%d%H%M%S")}_'
+    query_job = client.query(q, job_config, job_id_prefix=prefix)
+    result = query_job.result()
+
```

Comment:
I was going to request a log message on line 161 to tell me the query had finished, but it might be better to put the log message here.  It could be as simple as
```suggestion
    LOGGER.info(f"Query with prefix: `{prefix}` has completed")
```